wasm/src/debug_service.rs

```diff
@@ -31,12 +31,27 @@ impl DebugService {
         }
     }
 
-    pub fn load_source(&mut self, path: String, source: String, entry: Option<String>) -> String {
+    pub fn load_source(
+        &mut self,
+        path: String,
+        source: String,
+        target_profile: String,
+        entry: Option<String>,
+    ) -> String {
         let source_map = SourceMap::new(
             [(path.into(), source.into())],
             entry.as_deref().map(|value| value.into()),
         );
-        match Interpreter::new(true, source_map, qsc::PackageType::Exe, TargetProfile::Full) {
+        let target = match target_profile.as_str() {
+            "base" => TargetProfile::Base,
+            "full" => TargetProfile::Full,
+            _ => {
+                return render_error(qsc::interpret::stateful::Error::UnknownTarget(
+                    target_profile.to_string(),
```

Comment:
I think I'd rather sanitize this value before we call into WASM, and panic here if targetprofile is an invalid value. It doesn't feel worth adding a specific Interpreter error for, since getting the wrong value here would indicate a bug in _our_ code.

That being said, there are already 3 places where we read this value from settings and they all handle invalid values differently...

For display in the status bar, Defaults to "QIR:Invalid" for invalid value.
https://github.com/microsoft/qsharp/blob/b74867ceb788f490b95e6bb8a552faabc05c19ee/vscode/src/statusbar.ts#L58-L70

For configuring the language service, emits a warning but doesn't sanitize (this seems like a bug).
https://github.com/microsoft/qsharp/blob/b74867ceb788f490b95e6bb8a552faabc05c19ee/vscode/src/extension.ts#L228-L242

For QIR generation - just checks for "base" and ignores invalid values
https://github.com/microsoft/qsharp/blob/b74867ceb788f490b95e6bb8a552faabc05c19ee/vscode/src/qirGeneration.ts#L32-L33C13

This may be a good opportunity to refactor these into a utility function that sanitizes the value so we don't have to deal with handling invalid values everywhere?